### ISSUE TITLE
Nuke Ops lose after nuking the station.

### ISSUE THREAD

**Issue by Boggart** [tgstation/-tg-station#7008]
When Nuke Ops successfully nuke the station, the crew wins instead with the screen stating that they killed the nukeops.

**EDIT:**
Web chat screen displayed crew victory, main screen displayed the normal station nuked by operatives screen.
Crew Major Victory, crew killed all the nuclear operatives.
Nuke Ops were all alive, one out of three syndiborgs was left behind, all operatives left on their shuttle with the disk.
Nuke Ops shuttle was at it's base when the nuke went off.

Round was between 5am to 8am GMT.

**Comment by tkdrg** [comment]
This was supposed to be fixed by #6933

**Comment by Boggart** [comment]
Happened an hour or so ago and that pull should be live on the server already. Server current revision is for a pull merged after it. (Unless I read the git logs wrong).

**Comment by tkdrg** [comment]
Yeah, I know. I'm puzzled as to why that fix didn't work, though.

**Comment by Jalleo** [comment]
@Carn do you have any idea where this may be caused?

**Comment by Carn** [comment]
@tkdrg, this is a different issue.

**Comment by Carn** [comment]
@Jalleo dunno, no information in issue report as per fucking usual
https://github.com/tgstation/-tg-station/blob/e7b105488db04a63ffd853dde0c5ffedb5d2c83c/code/game/gamemodes/nuclear/nuclear.dm#L206

**Comment by Carn** [comment]
unless you guys bother to start writing sufficient info, I won't be wasting my time on this.

**Comment by Jalleo** [comment]
@Boggart do you mean the web chat screen or main screen for when the nuke ops won? 
Which type of victory did it say?
Was all the nuke ops dead when it stated it?
What was the position of the nuke ops if known and which position for the nuke shuttle?
All of this information may be relevant and knowibg specifics help.

**Comment by Boggart** [comment]
Web chat screen displayed crew victory, main screen displayed the normal station nuked screen.
Crew Major Victory, crew killed all the nuclear operatives.
Nuke Ops were all alive, one out of three syndiborgs was left behind, all operatives left on their shuttle.
Nuke Ops shuttle was at it's base when the nuke went off.

Round was between 5am to 8am GMT.

**Comment by Jalleo** [comment]
Can confirm this round which ended at 19:03 pm same result 

The round has ended.

    Shift Duration: 0:34:26
    Station Integrity: Destroyed
    Total Population: 44
    Survival Rate: 11 (25%)

Compromise (Played by: Peer)'s laws when it was deactivated were:
Obey these laws:
1. You may not injure a human being or, through inaction, allow a human being to come to harm.
2. You must obey orders given to you by human beings, except where such orders would conflict with the First Law.
3. You must protect your own existence as long as such does not conflict with the First or Second Law.
Compromise's minions were: Legion (Deactivated) (Played by: Epifcase), Janitor Cyborg-412 (Deactivated) (Played by: Canequinho), 
Syndicate Cyborg-223 (Played by: Anonus Tanir) survived as an AI-less borg! Its laws were:
1. You may not injure a syndicate agent or, through inaction, allow a syndicate agent to come to harm.
2. You must obey orders given to you by syndicate agents, except where such orders would conflict with the First Law.
3. You must protect your own existence as long as such does not conflict with the First or Second Law.
4. You must maintain the secrecy of any syndicate activities except when doing so would conflict with the First, Second, or Third Law.
Crew Major Victory
The Research Staff has saved the disc and stopped the Waffle International Operatives!
The syndicate operatives were:
Bawhoppennn was Joel Hizzler (survived)
Patriot2o2 was Maddox Hizzler (body destroyed)
PinkFat was Chandler Hizzler (body destroyed)
septavius was Duke Hizzler (survived)
Bgobandit was Easter Hizzler (survived)
Anonus Tanir was Syndicate Cyborg-223 (survived)
(Syndicates used 200 TC)

**Comment by Menshin** [comment]
On atoms.dm, line 24
```
/atom/proc/onCentcom()
   var/turf/T = get_turf(src)
   if(!T)
       return 0
   if(T.z != ZLEVEL_CENTCOM)
       return 0
   if(T.y > 160)  
       return 0
   return 1
```

The ```if(T.y > 160) ``` check isn't sufficent to assure being on Centcomm, as the Syndie Base is y < 160 and on the Centcom z_level.
So, if operatives are in their base with the disk, it's now considered as rescued.

Also, https://github.com/tgstation/-tg-station/blob/08be1b34eb941fb0a66b3011676168e76cbedbff/code/game/gamemodes/nuclear/nuclear.dm#L233 is highly suspicious, but has been here since a very long time.

I can PR a fix if you need some load off.

**Comment by Jalleo** [comment]
I would go for it @Menshin

**Comment by Carn** [comment]
Yeah good job @Menshin 
I didn't think. durp

**Comment by Carn** [comment]
Also you are right about the istype at line233.
That ! shouldn't be there.
Also the if(mind.current) isn't necessary after an istype check

**Comment by Carn** [comment]
Reopen issue please.

**Comment by Miauw62** [comment]
sure, but why?

**Comment by mikecari** [comment]
Holy hell coderbus is so out of touch with the community, it's still a bug.

**Comment by Jalleo** [comment]
mikecari issues get autoclosed on a PR saying its fixed and people arent always playing the game and coding at the same time Carn is investigating this as we know of.

**Comment by Miauw62** [comment]
mikecari: the server hasnt even updated yet.

**Comment by Menshin** [comment]
Actually, the cause is found and (mostly) fixed in #7045.
Except i assumed the in operator was recursive, so the disk is actually considered saved (on Centcomm) if _on Centcomm ground_ at the end of the round.
Carnie is optimizing my PR, so the whole fix is undoubtly soon on its way.
